doc/manuals/admin/build_source.md

```diff
@@ -25,15 +25,15 @@ commands that require root privilege):
 
 * Install the required libraries (except what needs to be taken from source, described in following steps).
 
-        sudo yum install boost-devel libcurl-devel gnutls-devel libgcrypt-devel openssl-devel libuuid-devel
+        sudo yum install boost-devel libcurl-devel gnutls-devel libgcrypt-devel openssl-devel libuuid-devel cyrus-sasl-devel
 
-* Install the Mongo Driver from source. The following procedure corresponds with default installation, if you want to include SASL and SSL support use [this alternative procedure](#building-mongodb-driver-with-sasl-and-ssl-support) instead.
+* Install the Mongo Driver from source.
 
         wget https://github.com/mongodb/mongo-cxx-driver/archive/legacy-1.1.2.tar.gz
         tar xfvz legacy-1.1.2.tar.gz
         cd mongo-cxx-driver-legacy-1.1.2
-        scons                                         # The build/linux2/normal/libmongoclient.a library is generated as outcome
-        sudo scons install --prefix=/usr/local        # This puts .h files in /usr/local/include/mongo and libmongoclient.a in /usr/local/lib
+        scons  --use-sasl-client --ssl                                        # The build/linux2/normal/libmongoclient.a library is generated as outcome
```

Comment:
Looking to the way the driver would compile, I have two doubts.

First, reading from http://mongodb.github.io/mongo-cxx-driver/legacy-v1/installation/#scons-options-when-compiling-the-c-driver:

> * `--ssl` Enables SSL support. You will need a compatible version of the SSL libraries available.The default authorization mechanism since MongoDB version 3.0 is SCRAM-SHA-1. If you want to use standard MongoDB authentication, you should compile with –ssl option for SCRAM-SHA-1 mechanism support.
> * `--use-sasl-client` Enables SASL, which MongoDB uses for the Kerberos authentication available on MongoDB Enterprise. You will need a compatible version of the SASL implementation libraries available. The Cyrus SASL libraries are what we test with, and are recommended.

I understand that `--ssl` is the one required by MongoDB default authentication in 3.4 ([SCRAM](https://docs.mongodb.com/v3.4/core/security-scram)) and `--use-sasl-client` is needed by [authentication mechanism only available in MongoDB Enterprise](https://docs.mongodb.com/v3.4/core/authentication-mechanisms-enterprise).  Is my interpretation correct?

Second, I wonder about clients not using authentication (which is a typical scenario in which the CB-MongoDB connection is secured using other means, e.g. level 3 firewalling) from the point of view of performance. I mean, it is fine that users wanting authentication pay a price for it (in terms of performance penalty of the SSL CB-MongoDB communication needed to authentication) but users not wanting authentication shouldn't be affected . Is the driver clever enough so even having being compiled using `--ssl` and `--use-sasl-client` clients not using authentication gets the same performance than if the driver would have been compiled without these options?

Could you elaborate on these aspects, please? Thanks!

Reply:
1. Your interpretation seems correct
2. I cannot elaborate on that performance impact. However if no authentication is used, I assume impact will be minimal...

Reply:
Asked at StackOverflow: https://stackoverflow.com/questions/51381582/does-mongodb-c-driver-compiled-with-authentication-options-penalize-clients-no

Reply:
Question still unanswered but shouldn't block PR on it (if in the future the question gets answered and modifications are needed we can do another PR if needed).

However, in order to merge the PR the following (or a similar wording) should be added to the CHANGES_NEXT_RELEASE file:

```
- Hardening: Mongo driver now compiled using --use-sasl-client --ssl to enable proper DB authentication mechanisms
```

@mathi123 could you manage, please? Thanks!

Reply:
Fixed in 2c114d5